README.md
# Flutter Cupertino Date Picker

[[pub packages]](https://pub.dartlang.org/packages/flutter_cupertino_date_picker)
| [中文说明](./README_zh-cn.md)

Flutter cupertino date picker.

![Example][1]

## Usage

#### 1\. Depend

Add this to you package's `pubspec.yaml` file:

```yaml
dependencies:
  flutter_cupertino_date_picker: ^1.0.26+2
```

#### 2\. Install

Run command:

```bash
$ flutter packages get
```

#### 3\. Import

Import in Dart code:

```dart
import 'package:flutter_cupertino_date_picker/flutter_cupertino_date_picker.dart';
```

#### 4\. Display DatePicker

##### Bottom Sheet DatePicker

```dart
/// Display date picker in bottom sheet.
///
/// context: [BuildContext]
/// minDateTime: [DateTime] minimum date time
/// maxDateTime: [DateTime] maximum date time
/// initialDateTime: [DateTime] initial date time for selected
/// dateFormat: [String] date format pattern
/// locale: [DateTimePickerLocale] internationalization
/// pickerMode: [DateTimePickerMode] display mode: date(DatePicker)、time(TimePicker)、datetime(DateTimePicker)
/// pickerTheme: [DateTimePickerTheme] the theme of date time picker
/// onCancel: [DateVoidCallback] pressed title cancel widget event
/// onClose: [DateVoidCallback] date picker closed event
/// onChange: [DateValueCallback] selected date time changed event
/// onConfirm: [DateValueCallback] pressed title confirm widget event
DatePicker.showDatePicker(
  BuildContext context,
  DateTime minDateTime,
  DateTime maxDateTime,
  DateTime initialDateTime,
  String dateFormat,
  DateTimePickerLocale locale: DATETIME_PICKER_LOCALE_DEFAULT,
  DateTimePickerMode pickerMode: DateTimePickerMode.date,
  DateTimePickerTheme pickerTheme: DatePickerTheme.Default,
  DateVoidCallback onCancel,
  DateVoidCallback onClose,
  DateValueCallback onChange,
  DateValueCallback onConfirm,
});
```

##### DatePicker Widget

```dart
/// Display date picker widget.
///
/// minDateTime: [DateTime] minimum date time
/// maxDateTime: [DateTime] maximum date time
/// initialDateTime: [DateTime] initial date time for selected
/// dateFormat: [String] date format pattern
/// locale: [DateTimePickerLocale] internationalization
/// pickerTheme: [DateTimePickerTheme] the theme of date time picker
/// onCancel: [DateVoidCallback] pressed title cancel widget event
/// onChange: [DateValueCallback] selected date time changed event
/// onConfirm: [DateValueCallback] pressed title confirm widget event
DatePickerWidget({
  DateTime minDateTime,
  DateTime maxDateTime,
  DateTime initialDateTime,
  String dateFormat: DATETIME_PICKER_DATE_FORMAT,
  DateTimePickerLocale locale: DATETIME_PICKER_LOCALE_DEFAULT,
  DateTimePickerTheme pickerTheme: DatePickerTheme.Default,
  DateVoidCallback onCancel,
  DateValueCallback onChange,
  DateValueCallback onConfirm,
})
```

##### TimePicker Widget

```dart
/// Display time picker widget.
///
/// minDateTime: [DateTime] minimum date time
/// maxDateTime: [DateTime] maximum date time
/// initialDateTime: [DateTime] initial date time for selected
/// dateFormat: [String] date format pattern
/// locale: [DateTimePickerLocale] internationalization
/// pickerTheme: [DateTimePickerTheme] the theme of date time picker
/// minuteDivider: [int] minute restriction, e.g. 5: every 5th minute will be shown (0, 5, 10, 15 ...)
/// onCancel: [DateVoidCallback] pressed title cancel widget event
/// onChange: [DateValueCallback] selected date time changed event
/// onConfirm: [DateValueCallback] pressed title confirm widget event
TimePickerWidget({
  DateTime minDateTime,
  DateTime maxDateTime,
  DateTime initialDateTime,
  String dateFormat: DATETIME_PICKER_DATE_FORMAT,
  DateTimePickerLocale locale: DATETIME_PICKER_LOCALE_DEFAULT,
  DateTimePickerTheme pickerTheme: DatePickerTheme.Default,
  int minuteDivider: 1,
  DateVoidCallback onCancel,
  DateValueCallback onChange,
  DateValueCallback onConfirm,
})
```

##### DateTimePicker Widget

```dart
/// Display date time picker widget.
///
/// minDateTime: [DateTime] minimum date time
/// maxDateTime: [DateTime] maximum date time
/// initialDateTime: [DateTime] initial date time for selected
/// dateFormat: [String] date format pattern
/// locale: [DateTimePickerLocale] internationalization
/// pickerTheme: [DateTimePickerTheme] the theme of date time picker
/// onCancel: [DateVoidCallback] pressed title cancel widget event
/// onChange: [DateValueCallback] selected date time changed event
/// onConfirm: [DateValueCallback] pressed title confirm widget event
DateTimePickerWidget({
  DateTime minDateTime,
  DateTime maxDateTime,
  DateTime initialDateTime,
  String dateFormat: DATETIME_PICKER_DATE_FORMAT,
  DateTimePickerLocale locale: DATETIME_PICKER_LOCALE_DEFAULT,
  DateTimePickerTheme pickerTheme: DatePickerTheme.Default,
  DateVoidCallback onCancel,
  DateValueCallback onChange,
  DateValueCallback onConfirm,
})
```

#### 5\. DateTimePickerLocale

Support:

- en_us: English (EN) United States ***[Default locale]***
- ar: Arabic (ar)
- ar_eg: Arabic (ar) Egypt
- bn: Bengali (BN)
- bs: Bosnian (BS)
- de: German (DE)
- es: Spanish (ES)
- fr: French (FR)
- hr: Croatian (HR)
- hu: Hungarian (HU)
- in_id: Bahasa (IN) Indonesia
- it: Italian (IT)
- jp: Japanese (JP)
- ko: Korea (KO)
- nl: Dutch (NL)
- pt_br: Portuguese (PT) Brazil
- ro: Romanian (RO)
- ru: Russian (RU)
- sr_cyrl: Serbia (sr) Cyrillic
- sr_latn: Serbia (sr) Latin
- tr: Turkish (TR)
- uk: Ukraine (UK)
- vi: Vietnamese (VI) Vietnam
- zh_cn: Chinese (ZH) Simplified

##### Add more language

###### 1. Add language i18n file

Fork this project, add language file in `lib/src/i18n/` directory, e.g. `strings_en_us.dart`.

```dart
/// English (EN) United States
class _StringsEnUs extends _StringsI18n {
  const _StringsEnUs();

  @override
  String getCancelText() {
    // TODO return cancel widget's text
    return null;
  }

  @override
  String getDoneText() {
    // TODO return done widget's text
    return null;
  }

  @override
  List<String> getMonths() {
    // TODO return the array of month's full name [January, February ... December]
    return null;
  }

  @override
  List<String> getMonthsShort() {
    // TODO return the array of month's short name [Jan, Feb ... Dec]. If return null, will substring the full name (max length is 3)
    return null;
  }

  @override
  List<String> getWeeksFull() {
    // TODO return the array of week's full name [Monday, Tuesday ... Sunday]
    return null;
  }

  @override
  List<String> getWeeksShort() {
    // TODO return the array of week's short name [Mon, Tue ... Sun]
    return null;
  }
}
```

###### 2. Add Locale

Add language locale in `lib/src/i18n/date_picker_i18n.dart` file.

```dart
enum DateTimePickerLocale {
  /// English (EN) United States
  en_us,
}
```

###### 3. Add Locale-Language relationship

Add language-locale relationship in `lib/src/i18n/date_picker_i18n.dart` file.

```dart
const Map<DateTimePickerLocale, _StringsI18n> datePickerI18n = {
  DateTimePickerLocale.en_us: const _StringsEnUs(),
};
```

#### 6\. dateFormat

|Pattern|Meaning                |e.g.      |
|:----|:------------------------|:---------|
|yyyy |year                     |2019, 2020|
|yy   |year, 2 digits           |19, 20|
|MMMM |month                    |January(en_us), 01(zh_cn)|
|MMM  |month, abbreviated       |Jan(en_us), 01(zh_cn)|
|MM   |month, 2 digits          |01、11
|M    |month                    |1, 11|
|dd   |day in month, 2 digits   |05, 25|
|d    |day in month             |5, 25|
|EEEE |day of week              |Monday(en_us), 星期一(zh_cn)|
|EEE  |day of week, abbreviated |Mon(en_us), 周一(zh_cn)|
|HH   |hour (0~23), 2 digits    |04, 12, 20|
|H    |hour (0~23)              |4, 12, 20|
|mm   |minute, 2 digits         |05, 40|
|m    |minute                   |5, 40|
|ss   |second, 2 digits         |06, 55|
|s    |second                   |6, 55|
|yyyy年|format                   |2019年, 2020年|
|H时   |format                   |5时, 21时|

##### Date Format Separator

Support separator: `|,-/\._: `.

#### 7\. DateTimePickerTheme

```dart
/// DateTimePicker theme.
///
/// [backgroundColor] DatePicker's background color.
/// [cancelTextStyle] Default cancel widget's [TextStyle].
/// [confirmTextStyle] Default confirm widget's [TextStyle].
/// [cancel] Custom cancel widget.
/// [confirm] Custom confirm widget.
/// [title] Custom title widget. If specify a title widget, the cancel and confirm widgets will not display. Must set [titleHeight] value for custom title widget.
/// [showTitle] Whether display title widget or not. If set false, the default cancel and confirm widgets will not display, but the custom title widget will display if had specified one custom title widget.
/// [pickerHeight] The value of DatePicker's height.
/// [titleHeight] The value of DatePicker's title height.
/// [itemHeight] The value of DatePicker's column height.
/// [itemTextStyle] The value of DatePicker's column [TextStyle].
const DateTimePickerTheme({
  this.backgroundColor: DATETIME_PICKER_BACKGROUND_COLOR,
  this.cancelTextStyle,
  this.confirmTextStyle,
  this.cancel,
  this.confirm,
  this.title,
  this.showTitle: DATETIME_PICKER_SHOW_TITLE_DEFAULT,
  this.pickerHeight: DATETIME_PICKER_HEIGHT,
  this.titleHeight: DATETIME_PICKER_TITLE_HEIGHT,
  this.itemHeight: DATETIME_PICKER_ITEM_HEIGHT,
  this.itemTextStyle: DATETIME_PICKER_ITEM_TEXT_STYLE,
});
```

## Example

[Example sources](https://github.com/wuzhendev/flutter-cupertino-date-picker/tree/master/example)

[Example APK](https://raw.githubusercontent.com/wuzhendev/assets/master/flutter-datepicker/flutter_cupertino_date_picker_v1.0.8.apk)

![Example APK Download](https://github.com/wuzhendev/assets/blob/master/flutter-datepicker/date_picker_qrcode_v1.0.8.png?raw=true)

### DatePicker

![Example: DatePicker][1]

![Example: DatePickerWidget][2]

### TimePicker

![Example: TimePicker][3]

![Example: TimePickerWidget][4]

### DateTimePicker

![Example: DateTimePicker][5]

![Example: DateTimePickerWidget][6]

## License

```
Copyright 2018 wuzhen

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
```

[1]:https://github.com/wuzhendev/assets/blob/master/flutter-datepicker/flutter_date_picker_14.jpg?raw=true
[2]:https://github.com/wuzhendev/assets/blob/master/flutter-datepicker/flutter_date_picker_15.jpg?raw=true
[3]:https://github.com/wuzhendev/assets/blob/master/flutter-datepicker/flutter_date_picker_16.jpg?raw=true
[4]:https://github.com/wuzhendev/assets/blob/master/flutter-datepicker/flutter_date_picker_17.jpg?raw=true
[5]:https://github.com/wuzhendev/assets/blob/master/flutter-datepicker/flutter_date_picker_18.jpg?raw=true
[6]:https://github.com/wuzhendev/assets/blob/master/flutter-datepicker/flutter_date_picker_19.jpg?raw=true


CHANGELOG.md
## 1.0.26+2

- Update pt_br language.

## 1.0.26+1

- Update pt_br language.

## 1.0.26

- Modify the button widget of the custom title bar to support hidden title and button widget:

```dart
DatePicker.showDatePicker(
  pickerTheme: DateTimePickerTheme(
    showTitle: true,
    confirm: Text('custom Done', style: TextStyle(color: Colors.red)),
    cancel: null, // If a custom title bar is set, the widget(title/confirm/cancel) will not be displayed when the value is null
  ),
);
```

## 1.0.25

- Fixed bug: [Issues #89](https://github.com/dylanwuzh/flutter-cupertino-date-picker/issues/89)

## 1.0.24

- Added language support: Dutch (NL).

## 1.0.23+1

- Remove `keystore.properties`.

## 1.0.23

- Added date format separator: `/`.
- `DatePicker.showDatePicker`: check pickerMode is legal or not by date format.

## 1.0.22

- Support `onMonthChangeStartWithFirstDate` for DatePicker: When the selected month changes, whether to set the selected date as the first day of the month.

## 1.0.21

- Added language support: Bosnian (BS) and Croatian (HR).

## 1.0.20

- Added language support: Ukraine (UK).

## 1.0.19+1

- Update LICENSE file.

## 1.0.19

- Added language support: Serbia Cyrillic(DateTimePickerLocale.sr_cyrl) and Latin(DateTimePickerLocale.sr_latn).

## 1.0.18

- Added language support: Turkish (DateTimePickerLocale.tr).

## 1.0.17

- Added language support: Vietnamese (DateTimePickerLocale.vi).

## 1.0.16

- Added language support: French (DateTimePickerLocale.fr).

## 1.0.15+1

- Modified Indonesia's subtag: DateTimePickerLocale.in_id -> DateTimePickerLocale.id

## 1.0.15

- Added language support: Bahasa Indonesia (DateTimePickerLocale.in_id).

## 1.0.14

- Added language support: Arabic Egypt (DateTimePickerLocale.ar_eg).

## 1.0.13+1

- Fixed bug: `DateTimePickerMode.datetime` mode display error. Issues: [#91](https://github.com/dylanwuzh/flutter-cupertino-date-picker/issues/91)

## 1.0.13

- Compatible with Flutter `dev` channel.

## 1.0.12

- Fixed bug: when `onClose` is null, catch exception `Unhandled Exception: NoSuchMethodError: The method 'call' was called on null. Receiver: null`. Issues: [#40](https://github.com/wuzhendev/flutter-cupertino-date-picker/issues/40) and [#43](https://github.com/wuzhendev/flutter-cupertino-date-picker/issues/43).

## 1.0.11

- Added `minuteDivider` parameter.

## 1.0.10

- Added `onClose` callback.

## 1.0.9

- Added language support: Hungarian (DateTimePickerLocale.hu).

## 1.0.8

- Refactor i18n codes.

## 1.0.7

- Added language support: Italian (DateTimePickerLocale.it).

## 1.0.6

- Added language support: Korea (DateTimePickerLocale.ko).

## 1.0.5

- Added language support: German (DateTimePickerLocale.de).

## 1.0.4

- Added language support: Russian (DateTimePickerLocale.ru).

## 1.0.3

- Added language support: Japanese (DateTimePickerLocale.jp).

## 1.0.2

- Added language support: Arabic (DateTimePickerLocale.ar).

## 1.0.1

- Added language support: Bengali (DateTimePickerLocale.bn).

## 1.0.0

- Refactor code.
- Support `DatePicker`、`TimePicker`、`DateTimePicker`.
- Support `DatePickerWidget`、`TimePickerWidget`、`DateTimePickerWidget`.
- Upgrade date format.
- Add `DateTimePickerTheme`.
- Add `DateTimePickerLocale` for locale.

## 0.8.0

- Support set the value of minimum DateTime and maximum DateTime.

## 0.7.0

- Change days in a month when moth changes.

## 0.6.0

- Add `DatePickerWidget` widget for display DatePicker in page.

## 0.5.0

- Add `onChanged2(DateTime dateTime, List<int> selectedIndex)` callback.
- Add `onConfirm2(DateTime dateTime, List<int> selectedIndex)` callback.
- Add `minDateTime`.
- Add `maxDateTime`.
- Add `initialDateTime`.

## 0.4.0

- Add `onCancel` callback.

## 0.3.0

- Support Flutter v1.0.

## 0.2.0

- Add custom cancel and confirm button.

## 0.1.0

- Refactor locale message.

## 0.0.7

- Support language support: Portuguese Brazil (DateTimePickerLocale.pt_br).

## 0.0.6

- Support English months.

## 0.0.5

- Fix month and day in zh locale.

## 0.0.4

- Support i18n.

## 0.0.2

- Complete README file.

## 0.0.1

- Initial Release.


lib/date_picker_i18n.dart
import 'dart:math';

enum DateTimePickerLocale {
  /// English (EN) United States
  en_us,

  /// Chinese (ZH) Simplified
  zh_cn,

  /// Portuguese (PT) Brazil
  pt_br,

  /// Spanish (ES)
  es,

  /// Romanian (RO)
  ro,

  /// Bengali (BN)
  bn,

  /// Arabic (ar)
  ar
}

const DATETIME_PICKER_LOCALE_DEFAULT = DateTimePickerLocale.en_us;

/// Done widget's text
const Map<DateTimePickerLocale, String> DONE = {
  DateTimePickerLocale.en_us: 'Kies',
  DateTimePickerLocale.zh_cn: '确定',
  DateTimePickerLocale.pt_br: 'Feito',
  DateTimePickerLocale.bn: 'সম্পন্ন',
  DateTimePickerLocale.ar: 'تم',
};

/// Cancel widget's text
const Map<DateTimePickerLocale, String> CANCEL = {
  DateTimePickerLocale.en_us: 'Annuleer',
  DateTimePickerLocale.zh_cn: '取消',
  DateTimePickerLocale.pt_br: 'Cancelar',
  DateTimePickerLocale.bn: 'বাতিল',
   DateTimePickerLocale.ar: 'ألغاء',
};

const List<String> DEFAULT_MONTH = [
  "01",
  "02",
  "03",
  "04",
  "05",
  "06",
  "07",
  "08",
  "09",
  "10",
  "11",
  "12"
];

/// en_US
const List<String> EN_US_MONTHS = [
  "Januari",
  "Februari",
  "Maart",
  "April",
  "Mei",
  "Juni",
  "Juli",
  "Augustus",
  "September",
  "Oktober",
  "November",
  "December"
];

/// es
const List<String> ES_MONTHS = [
  "Enero",
  "Febrero",
  "Marzo",
  "Abril",
  "Mayo",
  "Junio",
  "Julio",
  "Agosto",
  "Septiembre",
  "Octubre",
  "Noviembre",
  "Diciembre"
];

/// pt_BR
const List<String> PT_BR_MONTHS = [
  "Janeiro",
  "Fevereiro",
  "Março",
  "Abril",
  "Maio",
  "Junho",
  "Julho",
  "Agosto",
  "Setembro",
  "Outubro",
  "Novembro",
  "Dezembro"
];

/// ro
const List<String> RO_MONTHS = [
  "Ianuarie",
  "Februarie",
  "Martie",
  "Aprilie",
  "Mai",
  "Iunie",
  "Iulie",
  "August",
  "Septembrie",
  "Octombrie",
  "Noiembrie",
  "Decembrie"
];

/// ro
const List<String> BN_MONTHS = [
  "জানুয়ারী",
  "ফেব্রুয়ারি",
  "মার্চ",
  "এপ্রিল",
  "মে",
  "জুন",
  "জুলাই",
  "অগাস্ট",
  "সেপ্টেম্বর",
  "অক্টোবর",
  "নভেম্বর",
  "ডিসেম্বর"
];

/// ar
const List<String> AR_MONTHS = [
  "كانون الثاني",
  "شباط",
  "آذار",
  "نيسان",
  "أيار",
  "حزيران",
  "تموز",
  "آب",
  "أيلول",
  "تشرين الأول",
  "تشرين الثاني",
  "كانون الأول"
];


const Map<DateTimePickerLocale, List<String>> MONTHS = {
  DateTimePickerLocale.en_us: EN_US_MONTHS,
  DateTimePickerLocale.pt_br: PT_BR_MONTHS,
  DateTimePickerLocale.es: ES_MONTHS,
  DateTimePickerLocale.ro: RO_MONTHS,
  DateTimePickerLocale.bn: BN_MONTHS,
  DateTimePickerLocale.ar: AR_MONTHS,
};

/// en_US weeks with full name
const List<String> EN_US_WEEKS_FULL = [
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday",
  "Sunday",
];

/// en_US weeks with short name
const List<String> EN_US_WEEKS_SHORT = [
  "Mon",
  "Tue",
  "Wed",
  "Thur",
  "Fri",
  "Sat",
  "Sun",
];

/// es 西班牙语
const List<String> ES_WEEKS = [
  "Lunes",
  "Martes",
  "Miercoles",
  "Jueves",
  "Viernes",
  "Sabado",
  "Domingo",
];

/// pt_BR 葡萄牙语
const List<String> PT_BR_WEEKS = [
  "Segunda",
  "Terca",
  "Quarta",
  "Quinta",
  "Sexta",
  "Sabado",
  "Domingo",
];

/// zh_CN  weeks with full name
const List<String> ZH_CN_WEEKS_FULL = [
  "星期一",
  "星期二",
  "星期三",
  "星期四",
  "星期五",
  "星期六",
  "星期日",
];

/// zh_CN  weeks with short name
const List<String> ZH_CN_WEEKS_SHORT = [
  "周一",
  "周二",
  "周三",
  "周四",
  "周五",
  "周六",
  "周日",
];

/// bn weeks with full name
const List<String> BN_WEEKS_FULL = [
  "রবিবার",
  "সোমবার",
  "মঙ্গলবার",
  "বুধবার",
  "বৃহস্পতিবার",
  "শুক্রবার",
  "শনিবার",
];

/// bn weeks with short name
const List<String> BN_WEEKS_SHORT = [
  "রবি",
  "সোম",
  "মঙ্গল",
  "বুধ",
  "বৃহস্পতি",
  "শুক্র",
  "শনি",
];


/// ar weeks 
const List<String> AR_WEEKS = [
  "الأثنين",
  "الثلاثاء",
  "الأربعاء",
  "الخميس",
  "الجمعه",
  "السبت",
  "الأحد",
];

const Map<DateTimePickerLocale, List<String>> WEEKS_FULL = {
  DateTimePickerLocale.en_us: EN_US_WEEKS_FULL,
  DateTimePickerLocale.zh_cn: ZH_CN_WEEKS_FULL,
  DateTimePickerLocale.pt_br: PT_BR_WEEKS,
  DateTimePickerLocale.es: ES_WEEKS,
  DateTimePickerLocale.bn: BN_WEEKS_FULL,
  DateTimePickerLocale.ar: AR_WEEKS,

};

const Map<DateTimePickerLocale, List<String>> WEEKS_SHORT = {
  DateTimePickerLocale.en_us: EN_US_WEEKS_SHORT,
  DateTimePickerLocale.zh_cn: ZH_CN_WEEKS_SHORT,
  DateTimePickerLocale.bn: BN_WEEKS_SHORT,
};

class DatePickerI18n {
  /// Get done button text
  static String getLocaleDone(DateTimePickerLocale locale) {
    return DONE[locale] ?? DONE[DATETIME_PICKER_LOCALE_DEFAULT];
  }

  /// Get cancel button text
  static String getLocaleCancel(DateTimePickerLocale locale) {
    return CANCEL[locale] ?? DONE[DATETIME_PICKER_LOCALE_DEFAULT];
  }

  /// Get locale month array
  static List<String> getLocaleMonths(DateTimePickerLocale locale) {
    return MONTHS[locale] ?? DEFAULT_MONTH;
  }

  /// Get locale week array
  static List<String> getLocaleWeeks(DateTimePickerLocale locale,
      [bool isFull = true]) {
    if (isFull) {
      return WEEKS_FULL[locale] ?? WEEKS_FULL[DATETIME_PICKER_LOCALE_DEFAULT];
    }

    if (WEEKS_SHORT[locale] != null) {
      return WEEKS_SHORT[locale];
    }
    if (WEEKS_FULL[locale] != null) {
      return WEEKS_FULL[locale]
          .map((item) => item.substring(0, min(3, item.length)))
          .toList();
    }
    return WEEKS_FULL[DATETIME_PICKER_LOCALE_DEFAULT]
        .map((item) => item.substring(0, min(3, item.length)))
        .toList();
  }
}
